src/pip/_internal/locations/base.py

```diff
@@ -12,7 +11,9 @@
 USER_CACHE_DIR = appdirs.user_cache_dir("pip")
 
 # FIXME doesn't account for venv linked to global site-packages
-site_packages: typing.Optional[str] = sysconfig.get_path("purelib")
+_site_packages = sysconfig.get_path("purelib")
+assert _site_packages is not None
+site_packages = _site_packages
```

Comment:
This can *theoratically* be None, and pip should not crash on import-time IMO. I have some refactorings for this included in #10511.

Reply:
I see. I've restored `mypy: strict-optional=False` in `misc.py`. It can be revisited once that other work is finished.